### DISCUSSION TITLE
log4j vulnerability validation

### DISCUSSION THREAD

**Discussion by ghost** [pmd/pmd#3691] [Q&A] [answered]
Hi team, 
as we all know log4j vulnerability came recently and I am using pmd so need to know if pmd is using log4j internally or not.

https://logging.apache.org/log4j/2.x/security.html
https://nvd.nist.gov/vuln/detail/CVE-2021-45105
https://nvd.nist.gov/vuln/detail/CVE-2021-45046
https://nvd.nist.gov/vuln/detail/CVE-2021-44228

https://github.com/advisories/GHSA-jfh8-c2jp-5v3q
https://github.com/advisories/GHSA-7rjr-3q55-vv33
https://github.com/advisories/GHSA-p6xc-xr62-6r2g

**Comment by adangel** [comment, marked as answer]
PMD itself doesn't use log4j for logging (it uses java.util.logging).
We have a dependency to log4j though for test cases - so that typeresolution can resolve log4j loggers. This has been updated in the master branch for the next version (6.42.0) (https://github.com/pmd/pmd/commit/ae04e0a4b5f81daeb614ba697763bbcc7e88f0b2 and https://github.com/pmd/pmd/commit/76ccefa0f248839ff18b2961cb1c9aa2811aa4f0 and https://github.com/pmd/pmd/commit/1597ef3900869b2dd604339028f65c46b2c77aa8). But this is a **test** dependency, which means, it is not inherited.
